File: dto/user_record_dto.go
package dto

import "time"

type UserRecordDTO struct {
	ID             int64     `json:"id"`
	FirstName      string    `json:"firstName"`
	LastName       string    `json:"lastName"`
	Gender         string    `json:"gender"`
	IsTransferPin  bool      `json:"isTransferPin"`
	Locked         bool      `json:"locked"`
	LockedAt       time.Time `json:"lockedAt,omitempty"`
	ReferralCode   string    `json:"referralCode"`
	IsBlocked      bool      `json:"isBlocked"`
	BlockedDuration int64    `json:"blockedDuration,omitempty"`
	BlockedUntil   string    `json:"blockedUntil,omitempty"`
	BlockedReason  string    `json:"blockedReason,omitempty"`
	TotalRefs      string    `json:"totalRefs,omitempty"`
	Notifications  string    `json:"notifications,omitempty"`
	ReferralLink   string    `json:"referralLink,omitempty"`
	Photo          string    `json:"photo,omitempty"`
}

File: src/main/java/com/example/deposit/dto/UserRecordDTO.java
package com.example.deposit.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class UserRecordDTO {
    private Long id;
    private String firstName;
    private String lastName;
    private String gender;
    private boolean is_transfer_pin;
    private boolean locked;
    private LocalDateTime lockedAt;
    private String referralCode;
    private boolean isBlocked;
    private Long blockedDuration;
    private String blockedUntil;
    private String blockedReason;
    private String totalRefs;
    private String notifications;
    private String referralLink;
    private String photo;
}